core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -526,8 +529,23 @@ private void syncRemoteChangeEventToLocal(
         docConfig.getDocumentId(),
         remoteChangeEvent.getOperationType().toString()));
 
-    final DocumentVersionInfo currentRemoteVersionInfo = DocumentVersionInfo
-            .getRemoteVersionInfo(remoteChangeEvent.getFullDocument());
+    final DocumentVersionInfo currentRemoteVersionInfo;
```

Comment:
There's a bunch of these now. basically this guards us from pathological inputs.

Reply:
Nice. What do you mean pathological inputs? Just not a phrase I'm familiar with.

Reply:
I just mean inputs designed to go against how this code is supposed to function. So pathological in the mathematical sense. This is also for malicious input of a user trying to break sync.